crucible-mir/src/Mir/Trans.hs

```diff
@@ -1558,7 +1558,9 @@ evalPlaceProj ty pl@(MirPlace tpr ref meta) (M.PField idx _mirTy) = do
         case adt^.adtkind of
             Struct -> structFieldRef adt idx ref meta
             Enum _ -> mirFail $ "tried to access field of non-downcast " ++ show ty
-            Union -> mirFail $ "evalPlace (PField, Union) NYI"
+            Union -> do
+              info <- unionInfo adt idx
+              unionFieldRef info ref
```

Comment:
For consistency with `structFieldRef`/`enumFieldRef`, you could push the `unionInfo` call down into `unionFieldRef` (and `buildUnion`), making `unionInfo` purely an internal helper within `TransTy` like `structInfo` and `enumInfo` are